src/dynamics/solver/joint_constraint/ball_velocity_constraint.rs

```diff
@@ -40,17 +40,20 @@ impl BallVelocityConstraint {
         rb2: &RigidBody,
         joint: &BallJoint,
     ) -> Self {
-        let anchor1 = rb1.position * joint.local_anchor1 - rb1.world_com;
-        let anchor2 = rb2.position * joint.local_anchor2 - rb2.world_com;
+        let anchor_world1 = rb1.position * joint.local_anchor1;
+        let anchor_world2 = rb2.position * joint.local_anchor2;
+        let anchor1 = anchor_world1 - rb1.world_com;
+        let anchor2 = anchor_world2 - rb2.world_com;
```

Comment:
The names here are wildly inconsistent between files. It seems more common that `let anchor1 = position1 * self.local_anchor1;` (e.g. in `ball_position_constraint.rs`). Also sometimes the words `local` and `world` are part of the names (yay!) but sometimes not (😢 ). My approach is to try to make the smallest change possible to minimize conflicts.

Reply:
It seems what is here called `anchor1` is often called `r1`.